myapp/scheduler/stock_api_update.py

```diff
@@ -1,29 +1,26 @@
 from myapp.models import Stock
 from myapp import stock_api
+from django.db import transaction
 
 
-def update_top_stocks():
+@transaction.atomic
```

Comment:
I'm curious about this decorator is doing

Reply:
I read about `transaction.atomic`,  it's a good feature for aborting transactions in case any exception occurs; in your case, I think it's unneeded, because you already catching the exception, thus the `transaction.atomic` would not work.
see this for more information: 
https://stackoverflow.com/questions/34730385/django-rollback-save-with-transaction-atomic